modules/deploy/pages/deployment-option/cloud/gcp-private-service-connect.adoc

```diff
@@ -0,0 +1,317 @@
+--
+
+== Enable Private Service Connect on an existing BYOC cluster
+
+. In the Redpanda Cloud UI, go to the cluster overview and copy the cluster ID from the **Details** section.
++
+[,bash]
+----
+CLUSTER_ID=<cluster-id>
+----
+
+. Update the Redpanda Cloud Agent IAM role. This step is required only for clusters with xref:./vpc-byo-gcp.adoc[customer-managed resources].
++
+To allow the agent to create and manage the service attachment, add the following permissions to its IAM role:
++
+[,bash]
+----
+compute.forwardingRules.use
+compute.regionOperations.get
+compute.serviceAttachments.create
+compute.serviceAttachments.delete
+compute.serviceAttachments.get
+compute.serviceAttachments.list
+compute.serviceAttachments.update
+compute.subnetworks.use
+----
+
+. Make a request to the https://docs.api.redpanda.com/#patch-/v1beta1/clusters/-cluster.id-[`PATCH /v1beta1/clusters/{cluster.id}`] endpoint to update the cluster to include the newly-created Private Service Connect NAT subnet.
++
+[,bash]
+----
+export ACCEPT_LIST='[]'
+export PSC_NAT_SUBNET_NAME='<psc-nat-subnet-name>'
+export CLUSTER_PATCH_BODY=`cat << EOF
+{
+    "customer_managed_resources": {
+        "gcp": {
+            "psc_nat_subnet_name": "$PSC_NAT_SUBNET_NAME"
+        }
+    }
+}
+EOF`
+curl -v -X PATCH \
+-H "Content-Type: application/json" \
+-H "Authorization: Bearer $AUTH_TOKEN" \
+-d "$CLUSTER_PATCH_BODY" $PUBLIC_API_ENDPOINT/v1beta1/clusters/$CLUSTER_ID
+----
++
+Replace the following placeholder:
++
+`<psc-nat-subnet-name>`: The name of the Private Service Connect NAT subnet. Use the fully-qualified name, for example `"projects/<project>/regions/<region>/subnetworks/<subnet-name>"`.
+
+. Make a https://docs.api.redpanda.com/#patch-/v1beta1/clusters/-cluster.id-[`PATCH /v1beta1/clusters/{cluster.id}`] request to update the cluster to enable Private Service Connect.
++
+[,bash]
+----
+CLUSTER_PATCH_BODY=`cat << EOF
+{
+    "private_link": {
+        "enabled": true,
+        "gcp": {
+            "consumer_accept_list": <accept-list>
+        }
+    }
+}
+EOF`
+curl -v -X PATCH \
+-H "Content-Type: application/json" \
+-H "Authorization: Bearer $AUTH_TOKEN" \
+-d "$CLUSTER_PATCH_BODY" $PUBLIC_API_ENDPOINT/v1beta1/clusters/$CLUSTER_ID
+----
++
+Replace the following placeholder:
++
+`<accept-list>`: a JSON list specifying the projects from which incoming connections will be accepted. All other sources. For example, `[{"source": "consumer-project-ID-1"},{"source": "consumer-project-ID-2"}]`.
++
+Wait for the cluster to apply the new configuration (around 15 minutes). The Private Service Connect service attachment is available when the cluster update is complete. You can monitor the service attachment creation by running the following `gcloud` command and supplying the project ID:
++
+[,bash]
+----
+gcloud compute service-attachments list --project '<service-project-id>'
+----
+
+== Access Redpanda services through VPC endpoint
```

Comment:
Also, before this, we need to provide instructions on how to created the DNS configuration on their end. I'm sorry I missed this during my reviews.

To create the DNS zone:
```sh
gcloud dns --project=<GCP Project ID> managed-zones create <DNS zone name> --description="<description>" --dns-name="<DNS Zone from the UI>" --visibility="private" --networks="<list of fully-qualified name of networks where the DNS zone will be visible>"
```
To create the wildcard DNS A record:
```sh
gcloud dns --project=rp-byoc-juan-0e38 record-sets create '*.<DNS Zone from the UI>' --zone="<DNS zone name>" --type="A" --ttl="300" --rrdatas="<PSC endpoint IP>"
```